Analysis/HTauTauAnalysis/HTauTauAnalysis.C

```diff
@@ -0,0 +1,342 @@
+	    double names_of_leadlep_variable[]     = {Lepton.Pt(), Lepton.Eta(), Lepton.E(), Lepton.Phi(), (double)lep_charge->at(goodlep_index), (double)lep_type->at(goodlep_index)};
+	    TString histonames_of_leadlep_variable[]={"hist_leadleptpt", "hist_leadlepteta", "hist_leadleptE", "hist_leadleptphi", "hist_leadleptch", "hist_leadleptID"};
+
+	    double names_of_tau_variable[] ={ HadTau.Pt(), HadTau.Eta(), HadTau.E(), HadTau.Phi(), (double)tau_nTracks->at(goodtau_index) };
+	    TString histonames_of_tau_variable[]={"hist_taupt", "hist_taueta", "hist_tauE", "hist_tauphi", "hist_tau_nTracks" };
+	     
+	    //Start to fill histograms : find the histogram array length
+	    int length_global = sizeof(names_of_global_variable)/sizeof(names_of_global_variable[0]);
+	    int length_leadlep = sizeof(names_of_leadlep_variable)/sizeof(names_of_leadlep_variable[0]);
+	    int length_tau = sizeof(names_of_tau_variable)/sizeof(names_of_tau_variable[0]);
+	    
+	    //Fill histograms
+	    for (int i=0; i<length_global; i++){
+	      FillHistogramsGlobal( names_of_global_variable[i], weight, histonames_of_global_variable[i]);
+	    }
+	    
+	    for (int i=0; i<length_leadlep; i++){
+	      FillHistogramsLeadlept( names_of_leadlep_variable[i], weight, histonames_of_leadlep_variable[i]);
+	    }
+		    
+	    for (int i=0; i<length_tau; i++){
+	      FillHistogramsTau( names_of_tau_variable[i], weight, histonames_of_tau_variable[i]);
+	    }
+		    	    
+	    // fill jets
+	    if (jet_n > 0){
+	      double names_of_jet_variable[]={jet_pt->at(goodjet_index), jet_eta->at(goodjet_index)};
+	      TString histonames_of_jet_variable[]={"hist_leadjet_pt","hist_leadjet_eta"};
+	      int length_leadjet = sizeof(names_of_jet_variable)/sizeof(names_of_jet_variable[0]);
+	      for (int i=0; i<length_leadjet; i++){
+		FillHistogramsLeadJet( names_of_jet_variable[i], weight, histonames_of_jet_variable[i]);
+	      }
+	    } // jet_n 
+	  } //loop
+	} 
+      }
+    }
+  }
+  
+  return kTRUE;
+}
+
+void HTauTauAnalysis::SlaveTerminate()
+{
+  TString option = GetOption();
+  bool is_data = option.Contains("data");
+  
+  hist_scale_factors->SetBinContent(0, xsec_SF);
+  hist_scale_factors->SetBinContent(1, totalSumOfWeights_SF);
+  hist_scale_factors->SetBinContent(2, filteff_SF);
+  hist_scale_factors->SetBinContent(3, kfac_SF);
+  
+}
+
+void HTauTauAnalysis::Terminate()
+{
+
+  cout << "--------------------------------------------------------------------------------------" << endl;
+  cout << "The scaling factors values are: " << endl;
+  cout << "xsec: " << xsec_SF << endl;
+  cout << "totalSumOfWeights: " << totalSumOfWeights_SF  << endl;
+  cout << "filteff: " << filteff_SF << endl;
+  cout << "kfac: " << kfac_SF << endl;
+  cout << "--------------------------------------------------------------------------------------" << endl;
+  
+  TString filename_option = GetOption();
+  printf("Writting with name option: %s \n", filename_option.Data());
```

Comment:
```suggestion
  std::cout << "Writing with name option: " << filename_option.Data() << std::endl;
```

Any reason this can't avoid `printf`?